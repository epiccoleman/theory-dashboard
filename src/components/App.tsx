import * as React from 'react'
import { useState } from 'react'
import TonicDropdown from './TonicDropdown'
import ModeDropdown from './ModeDropdown'
import Chords from './Chords'
import SecondaryDominants from './SecondaryDominants'
import { Mode } from "../lib/Theory"
import { Container, Segment, Menu } from "semantic-ui-react"

const App = () => {
    const [ tonic, setTonic ] = useState("C")
    const [ mode , setMode ] = useState(Mode.Major);

    return (
        <Container>
            <Segment>
                <Menu>
                    <Menu.Item>
                        Key Picker
                    </Menu.Item>
                    <Menu.Item>
                        Chords
                    </Menu.Item>
                    <Menu.Item>
                        Secondary Dominants
                    </Menu.Item>
                </Menu>
            </Segment>
            <Segment>
                <TonicDropdown value={tonic} onChange={(e, { value }) => setTonic(value)}/>
                <ModeDropdown value={mode} onChange={(e, { value }) => setMode(value)}/>
            </Segment>
            <Segment>
                <Chords tonic={tonic} mode={mode} />
            </Segment>
            <Segment>
                <SecondaryDominants tonic={tonic} mode={mode} />
            </Segment>
        </Container>
            // <pre>{JSON.stringify(keyData(tonic, mode), null, 2)}</pre>
    );
}

export default App;