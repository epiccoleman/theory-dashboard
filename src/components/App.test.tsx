import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect'

import App from "./App";

describe("<App />", () => {
  test("renders", async () => {
    const { container, asFragment } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});