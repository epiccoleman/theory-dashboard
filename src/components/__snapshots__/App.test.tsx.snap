// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
<div>
  <div
    class="ui container"
  >
    <div
      class="ui segment"
    >
      <div
        class="ui menu"
      >
        <div
          class="item"
        >
          Key Picker
        </div>
        <div
          class="item"
        >
          Chords
        </div>
        <div
          class="item"
        >
          Secondary Dominants
        </div>
      </div>
    </div>
    <div
      class="ui segment"
    >
      <div
        aria-expanded="false"
        class="ui fluid selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-atomic="true"
          aria-live="polite"
          class="text"
          role="alert"
        >
          C
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              F
            </span>
          </div>
          <div
            aria-checked="true"
            aria-selected="true"
            class="active selected item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              C
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              G
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              D
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              A
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              E
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              B
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Gb
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Db
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Ab
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Eb
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Bb
            </span>
          </div>
        </div>
      </div>
      <div
        aria-expanded="false"
        class="ui fluid selection dropdown"
        role="listbox"
        tabindex="0"
      >
        <div
          aria-atomic="true"
          aria-live="polite"
          class="text"
          role="alert"
        >
          Major
        </div>
        <i
          aria-hidden="true"
          class="dropdown icon"
        />
        <div
          class="menu transition"
        >
          <div
            aria-checked="true"
            aria-selected="true"
            class="active selected item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Major
            </span>
          </div>
          <div
            aria-checked="false"
            aria-selected="false"
            class="item"
            role="option"
            style="pointer-events: all;"
          >
            <span
              class="text"
            >
              Minor
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ui segment"
    >
      <div
        class="ui container"
      >
        Cmaj7 Dm7 Em7 Fmaj7 G7 Am7 Bm7b5
      </div>
    </div>
    <div
      class="ui segment"
    >
      <div
        class="ui container"
      >
         A7 B7 C7 D7 E7 
      </div>
    </div>
  </div>
</div>
`;
